libphal/phal_sbe.C

```diff
@@ -370,5 +379,150 @@ void threadStopProc(struct pdbg_target *proc)
 	}
 }
 
+/**
+ * @brief Implements the recovery action to RESTART_SBE obtained by
+ *      p10_extract_sbe_rc HWP.
+ *
+ * @param[in] proc processor target to operate on
+ */
+void restartSBERCHandler(struct pdbg_target *proc)
+{
+	log(level::INFO, "restartSBERCHandler on %s", pdbg_target_path(proc));
+}
+
+/**
+ * @brief Switch the boot seeprom and REIPL using backup boot seeprom.
+ *
+ * @param[in] proc processor target to operate on
+ */
+void switchBootSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "reiplUsingBKPBootSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+/**
+ * @brief Switch the measurment seeprom and REIPL using backup measurement
+ *        seeprom
+ * @param[in] proc processor target to operate on
+ */
+void switchMeasurementSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "switchMeasurementSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+void validatePrimaryProcTgt(struct pdbg_target *proc)
+{
+	validatePrimaryProcTgt(proc);
+	if (!isPrimaryProc(proc)) {
+		log(level::ERROR,
+		    "validatePrimaryProcTgt: Target is not primary processor");
+		throw pdbgError_t(exception::PDBG_TARGET_INVALID);
+	}
+
+	isTgtPresent(proc);
+}
+
+void sbeErrorHandler(struct pdbg_target *proc)
+{
+	try {
+		log(level::INFO, "sbeErrorHandler on %s",
+		    pdbg_target_path(proc));
+
+		// Validate input target is processor target.
+		validatePrimaryProcTgt(proc);
+
+		enum sbe_state sbeState = getState(proc);
+
+		if (sbeState == SBE_STATE_DEBUG_MODE ||
```

Comment:
Why do we need to check the SBE state before executing the HWP? Here, you are not executing any chip-ops, right?

If you need to check the SBE state then, what about [other states](https://github.com/open-power/pdbg/blob/master/libpdbg/libpdbg_sbe.h#L13)?